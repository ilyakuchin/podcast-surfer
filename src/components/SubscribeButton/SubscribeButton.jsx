import React from 'react';
import { connect } from 'react-redux';
import {
  addSubscription,
  removeSubscription
} from '../../redux/actions/Subscriptions/subscriptions';

function isSubscribed(subscriptions, currentPodcast) {
  if (subscriptions) {
    const index = subscriptions.indexOf(currentPodcast);
    return index !== -1;
  }
  return false;
}

export function SubscribeButton({
  subscriptions,
  currentPodcastName,
  addSubscription,
  removeSubscription
}) {
  return (
    <div>
      <button
        onClick={e => {
          e.preventDefault();
          if (!isSubscribed(subscriptions, currentPodcastName)) {
            addSubscription(currentPodcastName);
          } else {
            removeSubscription(currentPodcastName);
          }
        }}
        type='button'
      >
        {!isSubscribed(subscriptions, currentPodcastName)
          ? 'Subscribe'
          : 'Unsbscribe'}
      </button>
    </div>
  );
}

const mapStateToProps = state => {
  return {
    subscriptions: state.subscriptions,
    currentPodcastName: state.currentPodcast.name
  };
};

const dispatchToProps = dispatch => {
  return {
    addSubscription: name => dispatch(addSubscription(name)),
    removeSubscription: name => dispatch(removeSubscription(name))
  };
};

const ConnectedSubscribeButton = connect(
  mapStateToProps,
  dispatchToProps
)(SubscribeButton);

export default ConnectedSubscribeButton;
