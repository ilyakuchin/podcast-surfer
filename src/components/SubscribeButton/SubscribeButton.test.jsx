import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SubscribeButton } from './SubscribeButton';

configure({ adapter: new Adapter() });

describe('Test SubscribeButton component', () => {
  test('should render correctly', () => {
    expect(
      toJson(
        shallow(
          <SubscribeButton
            subscriptions={[]}
            currentPodcastUrl=''
            jwt=''
            updateSubscriptionsConnect={() => {}}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
