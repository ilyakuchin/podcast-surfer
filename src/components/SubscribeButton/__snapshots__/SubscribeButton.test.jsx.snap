// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SubscribeButton component should render correctly 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Subscribe
  </button>
</div>
`;
